### PR TITLE
[GEOT-7451] Upgrade org.json:json to 20230618

### PR DESCRIPTION
**Pull request by turingtestfail**
[GEOT-745](https://osgeo-org.atlassian.net/browse/GEOT-745) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geotools&utm_campaign=badge_info)

  

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [x] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [x] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

File: modules/unsupported/flatgeobuf/pom.xml [hunk 1]
```diff
@@ -1,48 +1,49 @@
...
+        <dependency>
+            <groupId>org.geotools</groupId>
+            <artifactId>gt-geojson-core</artifactId>
+            <version>${project.version}</version>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.json</groupId>
+            <artifactId>json</artifactId>
+            <version>220230618</version>
```
Comment (mprins) [inline modules/unsupported/flatgeobuf/pom.xml:40]: There appears to be a typo here

**Review by mprins — CHANGES_REQUESTED** [review]

File: modules/unsupported/flatgeobuf/pom.xml (hunk 1 above)
Comment (turingtestfail) [reply modules/unsupported/flatgeobuf/pom.xml]: Fixed

**Review by mprins — APPROVED** [review]

**Comment by mprins** [comment]
there does not seem to be a relation to the failing mapfish print; that looks like a dependency fetching issue in gradle

**Comment by mprins** [comment]
@turingtestfail you may need to create a manual backport for 30.x; as 30-RC was already tagged and released. I've added the "backport 30.x" label but the bot does not seem to pick that up after merge..

**Comment by aaime** [comment]
The backport to `30.x` failed:
```
The process '/usr/bin/git' failed with exit code 1
```
<details>
<summary>stderr</summary>

```
To https://github.com/geotools/geotools.git
 ! [rejected]              backport-4487-to-30.x -> backport-4487-to-30.x (non-fast-forward)
error: failed to push some refs to 'https://github.com/geotools/geotools.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. If you want to integrate the remote changes,
hint: use 'git pull' before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

```
</details>


To backport manually, run these commands in your terminal:
```bash
# Fetch latest updates from GitHub
git fetch
# Create a new working tree
git worktree add .worktrees/backport-30.x 30.x
# Navigate to the new working tree
cd .worktrees/backport-30.x
# Create a new branch
git switch --create backport-4487-to-30.x
# Cherry-pick the merged commit of this pull request and resolve the conflicts
git cherry-pick 2820bcd1bc2d1fd3a731eca81f30e1724d2b31b9
# Push it to GitHub
git push --set-upstream origin backport-4487-to-30.x
# Go back to the original working tree
cd ../..
# Delete the working tree
git worktree remove .worktrees/backport-30.x
```
Then, create a pull request where the `base` branch is `30.x` and the `compare`/`head` branch is `backport-4487-to-30.x`.

**Comment by aaime** [comment]
The backport to `28.x` failed:
```
The process '/usr/bin/git' failed with exit code 1
```
<details>
<summary>stderr</summary>

```
To https://github.com/geotools/geotools.git
 ! [rejected]              backport-4487-to-28.x -> backport-4487-to-28.x (non-fast-forward)
error: failed to push some refs to 'https://github.com/geotools/geotools.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. If you want to integrate the remote changes,
hint: use 'git pull' before pushing again.
hint: See the 'Note about fast-forwards' in 'git push --help' for details.

```
</details>


To backport manually, run these commands in your terminal:
```bash
# Fetch latest updates from GitHub
git fetch
# Create a new working tree
git worktree add .worktrees/backport-28.x 28.x
# Navigate to the new working tree
cd .worktrees/backport-28.x
# Create a new branch
git switch --create backport-4487-to-28.x
# Cherry-pick the merged commit of this pull request and resolve the conflicts
git cherry-pick 2820bcd1bc2d1fd3a731eca81f30e1724d2b31b9
# Push it to GitHub
git push --set-upstream origin backport-4487-to-28.x
# Go back to the original working tree
cd ../..
# Delete the working tree
git worktree remove .worktrees/backport-28.x
```
Then, create a pull request where the `base` branch is `28.x` and the `compare`/`head` branch is `backport-4487-to-28.x`.

**Pull request merged**